lib/client/api_matrix.es6.js

```diff
@@ -324,64 +325,64 @@ foam.CLASS({
       args: [
         {
-          name: 'browserKeys',
+          name: 'releaseKeys',
           typeName: 'StringArray',
-          documentation: `A string array contains valid browser keys.`,
+          documentation: `A string array contains valid release keys.`,
         },
       ],
       returns: {
         typeName: 'JSON',
         documentation: `A JSON of the form:
-          {
-            browserKey:
             {
-              total: An integer, the total number of APIs belongs to
-                  this browser.
-              proprietary: An integer, number of APIs that only exists
-                  in this browser.
-              fallBehind: An integer, number of APIs that exists in all
-                  other selected browsers but not this browser.
-            }
-          }.`,
+              <releaseKey>:
+              {
+                total: An integer, the total number of APIs belongs to
+                    this release.
+                proprietary: An integer, number of APIs that only exists
+                    in this release.
+                fallBehind: An integer, number of APIs that exists in all
+                    other selected releases but not this release.
+              }
+            }.`,
       },
-      code: function(browserKeys) {
-        return this.browserApiDAO
+      code: function(releaseKeys) {
+        return this.releaseApiDAO
           .where(this.mlang.IN(
-            this.BrowserWebInterfaceJunction.SOURCE_ID, browserKeys))
+            this.ReleaseWebInterfaceJunction.SOURCE_ID, releaseKeys))
           .select(this.mlang.GROUP_BY(
-            this.BrowserWebInterfaceJunction.TARGET_ID,
+            this.ReleaseWebInterfaceJunction.TARGET_ID,
             this.ArraySink.create()))
           .then((groups) => {
             let result = {};
-            let totalBrowsers = browserKeys.length;
+            let totalReleases = releaseKeys.length;
             let interfaceKeys = groups.groupKeys;
-            for (let i = 0; i < browserKeys.length; i++) {
-              result[browserKeys[i]] = {
+            for (let i = 0; i < releaseKeys.length; i++) {
+              result[releaseKeys[i]] = {
                 total: 0,
                 proprietary: 0,
                 fallBehind: 0,
               };
             }
             for (let i = 0; i < interfaceKeys.length; i++) {
               let interfaceKey = interfaceKeys[i];
-              let availableBrowsers = {};
-              let browsers = groups.groups[interfaceKey].a;
-              for (let j = 0; j < browsers.length; j++) {
-                result[browsers[j].sourceId].total++;
-                availableBrowsers[browsers[j].sourceId] = true;
-                if (browsers.length === 1) {
-                  result[browsers[j].sourceId].proprietary++;
+              let availableReleases = {};
+              let releases = groups.groups[interfaceKey].a;
+              for (let j = 0; j < releases.length; j++) {
```

Comment:
The main reason I still use for-i instead of foreach-item is performance: https://jsperf.com/for-i-vs-foreach-item/1

Foreach tends to be fine for tests code and code invoked rarely with small "n", but since some of the code in this project iterates over, e.g., all APIs in a browser release, I've been sticking with the performant option.